code/game/machinery/machinery.dm

```diff
@@ -259,6 +259,10 @@ Class Procs:
 	if(!interact_offline && (stat & NOPOWER))
 		return STATUS_CLOSE
 
+	if(user.is_necromorph())
```

Comment:
this is what advancedtooluser is for, don't put specific typechecks in generic code like this